app/forms/reactivate_profile_form.rb

```diff
@@ -10,8 +10,11 @@ class ReactivateProfileForm
   attr_reader :user
 
   def initialize(user, attrs = {})
+    options = { recovery_code: [] }.merge(attrs)
```

Comment:
I'd recommend `attrs[:recovery_code] ||= {}` rather than a merge like this, the case I'm imagining is that somehow, we get passted in `{ recovery_code: nil }` which will get through here and cause an error when trying to join below